### ISSUE TITLE
“Unable to connect to Home Assistant” - Worked fine before, haven’t changed anything, works in browser

### ISSUE THREAD

**Issue by roo-punch** [home-assistant/android#1241]
**Home Assistant Android version:**

3.0.2-full

**Android version:**

8.0.0

**Phone model:**

SM-G930V (Samsung Galaxy 7)

**Home Assistant version:**

Home Assistant 2020.12.0

**Last working Home Assistant release (if known):**

Issue started on prior Home Assistant release randomly, updated in an attempt to resolve.

**Description of problem:**

Home Assistant Android Companion App does not load when outside local network connected to the external URL. I previously always used the external URL, and it randomly stopped working, on two different Android devices. I've got HA sitting behind and nginx reverse proxy server. I have NAT loopback on a pfsense box. The external URL works fine, both in and outside of my local network, and it works fine on the Android devices from the browser, but not in the Home Assistant App. I tried installing on a third device and get the issue “ERR_CONNECTION_TIMED_OUT”. The same external URL loads fine in the browser on that device.

**Traceback (if applicable):**

```

```

**Screenshot of problem:**

![Screenshot_20201217-092333_Home Assistant](https://user-images.githubusercontent.com/59415822/102514722-0c1e4500-404a-11eb-8790-063fafe9f2d1.jpg)

**Additional information:**

**Comment by dshokouhi** [comment]
If you said it worked before than what has changed in your network? Nothing in the app has changed in terms of how it connects to HA.  If it started happening randomly as you say then more than likely the issue is on your end.  You can find some troubleshooting stpes here: https://companion.home-assistant.io/docs/troubleshooting/networking

If you still have issues you will need to be much more specific in your setup.

**Comment by roo-punch** [comment]
Thank you for taking the time to respond. Before posting this, I poured over nginx config, modsec, firewall and router. I was suspecting something was blocking the user agent. I read on one forum post someone had the issue with an expired certificate, which I had ruled out, since I wasn't getting a warning in incognito and it was supposed to be updated automatically by certbot. Turns out, it was an expired certificate.. I had port 80 blocked.

It was very helpful to know I should go back and double check everything.
